### ISSUE TITLE
EditorConfig Editor Support

### ISSUE THREAD

**Issue by junichi11** [welovecoding/editorconfig-netbeans#45]
Hi, 

I'm developing an editor plugin for EditorConfig [1].
If it's convenient for users to unify it into your plugin, I'll merge them.

What do you think?

Thanks.

[1] https://github.com/junichi11/netbeans-editorconfig-editor

**Comment by bennyn** [comment]
Hi @junichi11,

I am glad to hear that you are building an EditorConfig editor. This is great news! 

You have already influenced the development of our EditorConfig plugin (with your encoding and change line endings plugin). Therefore it would be a great honor for us if you will contribute code to our plugin.

Just send us a Pull Request. 
We can also make you a direct contributor of this project if it makes things easier for you.

By the way, we also planned syntax highlighting for `.editorconfig` files in our plugin with issue [#39](https://github.com/welovecoding/editorconfig-netbeans/issues/39).

Looking forward to your Pull Request,

Benny

**Comment by junichi11** [comment]
Hi Benny,

I'll do it ;)
But there are a few conflicts as the followings: 

1. mime-type (text/x-editorconfig)
2. icon 
![icon16](https://cloud.githubusercontent.com/assets/738383/6341224/9debeaf6-bc0b-11e4-9269-892e594e98e2.png)

If you want to use your's, I'll change my plugin's code. So please let me know that :)

> By the way, we also planned syntax highlighting for .editorconfig files in our plugin with issue #39.

My plugin provides support the followings:

- Syntax highlighting
- Toggle line comment (`#` or `;`) #29 
- Code templates

#### Code templates:
![nb-editorconfig-issue45-1](https://cloud.githubusercontent.com/assets/738383/6341355/d7118c2c-bc0c-11e4-9954-426baa049132.png)
![nb-editorconfig-issue45-2](https://cloud.githubusercontent.com/assets/738383/6341357/da1a5b74-bc0c-11e4-8a3b-762082bd5368.png)
![nb-editorconfig-issue45-3](https://cloud.githubusercontent.com/assets/738383/6341360/e14d91d6-bc0c-11e4-9b9e-0211bc0cc676.png)

Thnaks,
Junichi

**Comment by bennyn** [comment]
Hi Juinchi,

when we started the EditorConfig plugin we were unsure what MIME type we should use. We went for `text/plain+editorconfig` (https://github.com/welovecoding/editorconfig-netbeans/commit/3fb7b37f31321d071b37220f8302083abf14dc3d). Then we asked the community (https://github.com/editorconfig/editorconfig/issues/185) and they suggested to use something related to INI. INI files are recognized in NetBeans by the MIME type `text/x+ini`. Since the EditorConfig type is still quite new, it makes sense to mark it with the **x-** flag.

- So let's use `text/x-editorconfig`.

Regarding the EditorConfig logos we have tried a lot. We also got a high resolution version of it from @treyhunner. After all the experiments we found that the current [image](https://github.com/welovecoding/editorconfig-netbeans/blob/master/src/main/resources/com/welovecoding/netbeans/plugin/editorconfig/filetype/ec.png) looks best in NetBeans.

- So I would suggest to use our logo or are there problems with it?

Your syntax highlighting and code completion is brilliant! The users will be happy. 
I will change the used MIME type quickly.

Best from Berlin,

Benny

**Comment by junichi11** [comment]
> So I would suggest to use our logo or are there problems with it?

No problem. Just asked. I follow your plugin :)

BTW, I have a question and a request. 

- What's the license of this plugin?
- Could you add me to this plugin as a contributor if possible? (I create a new branch, then I'll sent a self PR) If you would like to avoid that, no problem :) I'll sent it from forked repository.

Thanks.

**Comment by bennyn** [comment]
**Regarding the license:** We have not defined a license yet. People can use our code as they like. I think for developers the most flexible license is MIT. I have seen that you use MIT license for [netbeans-editorconfig-editor](https://github.com/junichi11/netbeans-editorconfig-editor). So we can do the same if you want.

**Regarding the contribution:** We can add you as a direct contributor, no problem. But because we have the code on our organization ([welovecoding](https://github.com/welovecoding)), it will require you to become a member of our organization. I can add you if you want. 

**Just to give you an idea what our organization is about:**
Our organization is no company. It's more like a coding club. We host code in our organization which can be valuable for developers. We Love Coding is also a website where people can learn programming with free video tutorials (check www.welovecoding.com). We created apps for Android, iOS and Windows 8 to distrubute these videos. And people are really happy about it (got more than 300.000 app installations). 

Our next step is to setup a new website (design is already done) but first let's release version 1.0 of the NetBeans EditorConfig plugin. :grinning:

**Comment by junichi11** [comment]
Thanks for your answer!

> Regarding the license: We have not defined a license yet. People can use our code as they like. I think for developers the most flexible license is MIT. I have seen that you use MIT license for netbeans-editorconfig-editor. So we can do the same if you want.

I see. License will be needed if we publish this plugin on NetBeans Plugin Portal. My code have a license header. If you use a license other than MIT, I'll remove it once when I merge plugin. Basically, I would like to follow your plugin's rule :smiley: 

> Regarding the contribution: We can add you as a direct contributor, no problem. But because we have the code on our organization (welovecoding), it will require you to become a member of our organization. I can add you if you want.

It's OK to become a member of your organization :+1:

**Comment by bennyn** [comment]
Hi @junichi11,

I checked http://choosealicense.com/ and according to their information the MIT license is best for us. But inside our plugin we are using [editorconfig-core-java](https://github.com/editorconfig/editorconfig-core-java) which uses the the Apache license. So we should double check if we can distribute our combined work with MIT license when including a project with Apache license. According to [a question on Quora](http://www.quora.com/Is-the-MIT-license-compatible-with-the-Apache-License-Version-2-APLv2) it should work. But we also have to check which licenses our Java dependencies use. Other EditorConfig plugins are using also MIT ([editorconfig-jetbrains](https://github.com/editorconfig/editorconfig-jetbrains/blob/master/LICENSE.txt)) or Simplified BSD ([editorconfig-jedit](https://github.com/editorconfig/editorconfig-jedit/blob/master/LICENSE)).

- So let's face this issue later when it comes to publishing the plugin. 

Regarding the organization, I sent you an invite so you can become a member of our organization. You should have access to modify this repository here. Please let me know if there are any issues with it. And thanks for the +1 on Facebook. :smiley:

**Comment by junichi11** [comment]
Hi Benny,

> So let's face this issue later when it comes to publishing the plugin.

:+1: 

> I sent you an invite so you can become a member of our organization.

Thanks a lot for adding me!

> And thanks for the +1 on Facebook.

I've done it a little ago ;)

**Comment by bennyn** [comment]
Hello and welcome to our organization! :octocat:  

Today, I have read a bit about BSD, Apache v2 license and MIT license. The answers on the Internet are very vague. However, I've found a few interesting things. The MIT license for example allows for distribution without contribution credits; BSD doesn't [[source]](https://programmers.stackexchange.com/questions/121998/mit-vs-bsd-vs-dual-license/122008#122008).

I think it would be nice to get contribution credits but we also have to make sure that our code can be used by the NetBeans IDE team to be integrated in NetBeans itself (if they think it's worth it). So I think that our main goals with a license should be:

- Make to code freely available to other developers
- Make sure that the NetBeans IDE team can integrate it in their source code
- Make sure that we can get some credits (if possible)

What's your feeling about these goals?

I will read some more articles before coming up with license soltions. I already found these nice posts:
- [ZDNet — Greg Stein: Three is enough](http://www.zdnet.com/article/greg-stein-three-is-enough/)
- [Google Open Source Blog — Standing Against License Proliferation](http://google-opensource.blogspot.de/2008/05/standing-against-license-proliferation.html)
- [O'Reilly Media — The MIT, BSD, Apache, and Academic Free Licenses](http://www.oreilly.com/openbook/osfreesoft/book/ch02.pdf)
